src/main/java/com/yahoo/bullet/bql/query/TypeChecker.java

```diff
@@ -280,13 +280,10 @@
             case GREATER_THAN_OR_EQUALS:
             case LESS_THAN:
             case LESS_THAN_OR_EQUALS:
-                if (!Type.isNumeric(leftType)) {
-                    errors.add(makeErrorOnly(node, QueryError.BINARY_LHS_NOT_NUMERIC, node, leftType));
-                }
-                if (!Type.isNumeric(rightType)) {
-                    errors.add(makeErrorOnly(node, QueryError.BINARY_RHS_NOT_NUMERIC, node, rightType));
+                if (!(Type.isNumeric(leftType) && Type.isNumeric(rightType)) && !(leftType == Type.STRING && rightType == Type.STRING)) {
```

Comment:
Can we write a `isComparable(Type l, Type r)` and reuse for these checks? That function can do the check for if l is numeric and then r must be numeric or l is string and then r must be string. You can use that for both lhs and rhs or lhs and rhs.subtype

Reply:
The primitiveList check should of course happen outside this method